packages/cozy-harvest-lib/src/models/KonnectorJob.js

```diff
@@ -134,24 +141,113 @@ export class KonnectorJob {
     return this.account
   }
 
-  getKonnectorPolicy() {
-    return findKonnectorPolicy(this.konnector)
+  async saveTwoFARequest(twoFARequestOptions) {
+    this.setState({ status: WAITING_TWOFA })
+    this.triggerEvent(TWO_FA_REQUEST_EVENT, twoFARequestOptions)
+    try {
+      const account = accounts.updateTwoFAState(
+        this.account,
+        twoFARequestOptions
+      )
+      await this.saveAccount(account)
+    } catch (error) {
+      // eslint-disable-next-line no-console
+      console.error(error)
+      this.setState({ status: ERRORED, error })
+    }
+  }
+
+  waitForTwoFA() {
+    logger.info('Waiting for two FA')
+    return new Promise(rawResolve => {
+      const accountId = this.account._id
+      assert(accountId, 'Cannot wait for two fa on account without id')
+
+      const finishWaiting = () => {}
+
+      const resolve = () => {
+        this.realtime.unsubscribe(
+          'updated',
+          ACCOUNTS_DOCTYPE,
+          accountId,
+          handleAccountUpdate
+        )
+        rawResolve()
+      }
+
+      const handleAccountUpdate = account => {
+        if (account && account.twoFACode) {
+          resolve()
+        }
+      }
+
+      const updateFromCustomPolicy = () => {
+        resolve()
+      }
+
+      this.twoFAWaiters.push(updateFromCustomPolicy)
+
+      this.realtime.subscribe(
+        'updated',
+        ACCOUNTS_DOCTYPE,
+        accountId,
+        handleAccountUpdate
+      )
+    })
   }
 
   /**
-   * "Sends" 2FA Code, saves it into account
+   * Saves and updates internal account
+   */
+  async saveAccount(updatedAccount) {
+    logger.debug('Saving account')
+    this.account = await saveAccount(
+      this.client,
+      this.konnector,
+      updatedAccount
+    )
+    logger.info('Saved account')
+    return this.account
+  }
+
+  flushTwoFAWaiters() {
+    logger.debug(`Flushing ${this.twoFAWaiters.length} two fa waiters`)
+    for (const w of this.twoFAWaiters) {
+      w()
```

Comment:
Nit: I think renaming `w` to `waiter` would make reading easier

Reply:
Renamed to callback since it is called.